### PR TITLE
[HTTPConnectionPool] Fix request timer scheduling

### PR DESCRIPTION
**Pull request by fabianfett**
### Motivation

If the allocator puts a request at the same address where another request has lived before and we ran into a timer race before, we run into an assertion. This can be seen in test runs here:

https://ci.swiftserver.group/job/async-http-client-swift55-prb/137/console

### Changes

- Remove `timerLock` from `HTTPConnectionPool`

### Result

- No more race conditions. One lock is easier than two.

### CODE REVIEW

**Comment by fabianfett** [comment]
`swift-nightly` failing because of Sendable requirements.

File: Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift [hunk 1]
```diff
@@ -282,49 +323,29 @@ final class HTTPConnectionPool {
...
-    }
-
-    private func cancelRequestTimeouts(_ requests: [Request]) {
-        let scheduled = self.timerLock.withLock {
-            requests.compactMap {
-                self._requestTimer.removeValue(forKey: $0.id)
-            }
-        }
-        scheduled.forEach { $0.cancel() }
+        cancelTimer.cancel()
```
Comment (Lukasa) [inline Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift:348]: Should we do these `cancel`s outside the lock?

Comment (fabianfett) [reply Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift:348]: Good idea!

Comment (fabianfett) [reply Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift:348]: Will do at a later point, tracked here: https://github.com/swift-server/async-http-client/issues/439

**Review by Lukasa — APPROVED** [review]

**Pull request merged**